source/common/grpc/async_client_manager_impl.h

```diff
@@ -50,18 +54,64 @@ class AsyncClientManagerImpl : public AsyncClientManager {
   AsyncClientManagerImpl(Upstream::ClusterManager& cm, ThreadLocal::Instance& tls,
                          TimeSource& time_source, Api::Api& api, const StatNames& stat_names);
 
+  RawAsyncClientSharedPtr
+  getOrCreateRawAsyncClient(const envoy::config::core::v3::GrpcService& config, Stats::Scope& scope,
+                            bool skip_cluster_check) override {
+    RawAsyncClientSharedPtr client;
```

Comment:
move these implementations to a `.cc ` file. Some of these will expand to a lot of code which will have to be recompiled at every use and then (hopefully) e-duped by the linker.

Generally we should online inline if the impl is trivial (e..g `return member_var;`) or required, due to templating.

Reply:
Thanks, will do.